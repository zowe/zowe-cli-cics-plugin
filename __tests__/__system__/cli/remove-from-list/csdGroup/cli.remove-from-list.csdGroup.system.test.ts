/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../../__src__/doc/ITestPropertiesSchema";

let TEST_ENVIRONMENT: ITestEnvironment<ITestPropertiesSchema>;
let regionName: string;
let csdGroup: string;
let host: string;
let port: number;
let user: string;
let password: string;
let protocol: string;
let rejectUnauthorized: boolean;
describe("CICS remove-from-list csdGroup command", () => {

    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "remove_from_list_csdGroup",
            installPlugin: true,
            tempProfileTypes: ["cics"]
        });
        csdGroup = TEST_ENVIRONMENT.systemTestProperties.cmci.csdGroup;
        regionName = TEST_ENVIRONMENT.systemTestProperties.cmci.regionName;
        host = TEST_ENVIRONMENT.systemTestProperties.cics.host;
        port = TEST_ENVIRONMENT.systemTestProperties.cics.port;
        user = TEST_ENVIRONMENT.systemTestProperties.cics.user;
        password = TEST_ENVIRONMENT.systemTestProperties.cics.password;
        protocol = TEST_ENVIRONMENT.systemTestProperties.cics.protocol;
        rejectUnauthorized = TEST_ENVIRONMENT.systemTestProperties.cics.rejectUnauthorized;
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should be able to display the help", () => {
        const output = runCliScript(__dirname + "/__scripts__/remove_from_list_csdGroup_help.sh", TEST_ENVIRONMENT, []);
        expect(output.stderr.toString()).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toMatchSnapshot();
    });

    it("should get a syntax error if csdGroup name is omitted", () => {
        const output = runCliScript(__dirname + "/__scripts__/remove_from_list_csdGroup.sh", TEST_ENVIRONMENT,
            ["", "FAKELIST", "FAKERGN"]);
        const stderr = output.stderr.toString();
        expect(stderr).toContain("Syntax");
        expect(stderr).toContain("Missing Positional Argument");
        expect(stderr).toContain("name of the CSD Group");
        expect(output.status).toEqual(1);
    });

    it("should get a syntax error if list name is omitted", () => {
        const output = runCliScript(__dirname + "/__scripts__/remove_from_list_csdGroup.sh", TEST_ENVIRONMENT,
            ["FAKEGRP", "", "FAKERGN"]);
        const stderr = output.stderr.toString();
        expect(stderr).toContain("Syntax");
        expect(stderr).toContain("Missing Positional Argument");
        expect(stderr).toContain("name of the CSD List");
        expect(output.status).toEqual(1);
    });

    it("should be able to successfully add and remove a csdGroup to/from a list with basic options", async () => {
        const dummyList = "TESTLIST";
        let output = runCliScript(__dirname + "/../../add-to-list/csdGroup/__scripts__/add_to_list_csdGroup.sh", TEST_ENVIRONMENT,
            [csdGroup, dummyList,  regionName]);
        let stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");

        output = runCliScript(__dirname + "/__scripts__/remove_from_list_csdGroup.sh", TEST_ENVIRONMENT,
            [csdGroup, dummyList,  regionName]);
        stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");
    });

    it("should be able to successfully add and remove a csdGroup to/from a list with profile options", async () => {
        const dummyList = "TESTLIST";
        let output = runCliScript(__dirname + "/../../add-to-list/csdGroup/__scripts__/add_to_list_csdGroup_fully_qualified.sh", TEST_ENVIRONMENT,
            [csdGroup,
                dummyList,
                regionName,
                host,
                port,
                user,
                password,
                protocol,
                rejectUnauthorized]);
        let stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");

        output = runCliScript(__dirname + "/__scripts__/remove_from_list_csdGroup_fully_qualified.sh", TEST_ENVIRONMENT,
            [csdGroup,
                dummyList,
                regionName,
                host,
                port,
                user,
                password,
                protocol,
                rejectUnauthorized]);
        stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");
    });
});
