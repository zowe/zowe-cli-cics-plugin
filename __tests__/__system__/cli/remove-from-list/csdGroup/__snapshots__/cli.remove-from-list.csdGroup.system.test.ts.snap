// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CICS remove-from-list csdGroup command should be able to display the help 1`] = `
"
 COMMAND NAME
 ------------

   csdGroup | csdg

 DESCRIPTION
 -----------

   Remove a CSD Group from a CICS CSD List.

 USAGE
 -----

   zowe cics remove-from-list csdGroup <name> <csdList> [options]

 POSITIONAL ARGUMENTS
 --------------------

   name		 (string)

      The name of the CSD Group to remove. The maximum length of the CSD Group name is
      eight characters

   csdList		 (string)

      The name of the CSD List to remove the group from. The maximum length of the CSD
      List name is eight characters

 OPTIONS
 -------

   --region-name (string)

      The CICS region name to which to remove the CSD Group from the CSD List

   --cics-plex (string)

      The name of the CICSPlex to which to remove the CSD Group from the CSD List

 CICS CONNECTION OPTIONS
 -----------------------

   --host | -H (string)

      The CICS server host name.

   --port | -P (number)

      The CICS server port.

      Default value: 443

   --user | -u (string)

      Mainframe (CICS) user name, which can be the same as your TSO login.

   --password | --pw (string)

      Mainframe (CICS) password, which can be the same as your TSO password.

   --reject-unauthorized | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --protocol | -o (string)

      Specifies CMCI protocol (http or https).

      Default value: https
      Allowed values: http, https

 PROFILE OPTIONS
 ---------------

   --cics-profile | --cics-p (string)

      The name of a (cics) profile to load for this command execution.

   --base-profile | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --token-type | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value | --tv (string)

      The value of the token to pass to the API.

   --cert-file (local file path)

      The file path to a certificate file to use for authentication

   --cert-key-file (local file path)

      The file path to a certificate key file to use for authentication

 GLOBAL OPTIONS
 --------------

   --show-inputs-only (boolean)

      Show command inputs and do not run the command

   --response-format-json | --rfj (boolean)

      Produce JSON formatted data from a command

   --help | -h (boolean)

      Display help text

   --help-web | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Remove the CSD Group MYGRP from the CSD List MYLIST in the
   region named MYREG:

      $ zowe cics remove-from-list csdGroup MYGRP MYLIST --region-name MYREG

"
`;
